invenio_app_rdm/theme/views.py

```diff
@@ -105,23 +105,22 @@ def deposits_edit(pid_value):
         # Dereference relations (languages, licenses, etc.)
         draft._record.relations.dereference()
         serializer = UIJSONSerializer()
-        _record = serializer.serialize_object_to_dict(draft._record)
+        record = serializer.serialize_object_to_dict(draft.to_dict())
+
         # TODO: get the `is_published` field when reading the draft
         from invenio_pidstore.errors import PIDUnregistered
         try:
-            _ = service.draft_cls.pid.resolve(
-                    pid_value, registered_only=True)
-            _record["is_published"] = True
+            service.draft_cls.pid.resolve(pid_value, registered_only=True)
```

Comment:
minor: I personally think is useful to have the `_` when it returns something and it is not used. Also, the fact that the return from the resolver is not used might turn into inconsistencies:

- Do something
- Resolve
- Return the original record/draft (not the resolved one)
- The user sees the original (Correct), but the DB/ES one for some reason did not register the action/change.
- Next time the resolved one is taken... There was a change not saved/performed

Maybe too much of a corner case...

Reply:
My take is when a variable is assigned, it's so it can be reused. Since it's not reused it doesn't need to be there (minimalist value) and it's less confusing (was it supposed to be reused? was it supposed to be called `_` because of internationalization? ...). Not sure I follow the corner case :sweat_smile:  - assigning the variable doesn't change what resolve returns...

Reply:
well... my point was that actually should be reused to avoid the corner case... but ok... let's hope we don't bump into it :)